src/pip/__main__.py

```diff
@@ -13,6 +13,12 @@
     path = os.path.dirname(os.path.dirname(__file__))
     sys.path.insert(0, path)
 
+# Remove '' and current working directory from the first entry
+# of sys.path, if present to avoid using current directory
+# in pip commands
+if sys.path[0] in ('', os.getcwd()):
```

Comment:
nit: since we are checking the first element of `sys.path` it would make more sense to do it before the previous `if __package__ == '':` that could put the wheel as first element of `sys.path`

Reply:
HI @xavfernandez 

I have fixed the location of the check to be before `if __package__ == '':`. Please take a look